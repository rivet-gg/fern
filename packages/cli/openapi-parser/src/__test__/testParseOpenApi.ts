import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
