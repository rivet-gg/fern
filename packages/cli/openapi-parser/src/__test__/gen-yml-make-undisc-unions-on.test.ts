import { testParseOpenAPI } from "./testParseOpenApi";

describe("open api parser", () => {
    testParseOpenAPI("gen-yml-make-undisc-unions", "openapi.json", undefined, {
        audiences: [],
        shouldUseTitleAsName: true,
        shouldUseUndiscriminatedUnionsWithLiterals: true
    });
});
