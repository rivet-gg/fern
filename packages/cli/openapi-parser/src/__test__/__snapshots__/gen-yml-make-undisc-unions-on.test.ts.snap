// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api parser gen-yml-make-undisc-unions parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": "The Batch API provides access to Hume models through an asynchronous job-based interface. You can submit a job to have many different files processed in parallel. The status of a job can then be checked with the job ID. Email notifications are available to alert on completed jobs.",
  "endpoints": [],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [],
  "schemas": {
    "UnionJob": {
      "type": "oneOf",
      "value": {
        "availability": null,
        "description": null,
        "generatedName": "UnionJob",
        "groupName": null,
        "nameOverride": null,
        "schemas": [
          {
            "availability": null,
            "description": null,
            "generatedName": "ComponentsSchemasUnionJobJobEmbeddingGeneration",
            "groupName": null,
            "nameOverride": null,
            "schema": "UnionJob_Job<EmbeddingGeneration>",
            "type": "reference",
          },
          {
            "availability": null,
            "description": null,
            "generatedName": "ComponentsSchemasUnionJobJobInference",
            "groupName": null,
            "nameOverride": null,
            "schema": "UnionJob_Job<Inference>",
            "type": "reference",
          },
          {
            "availability": null,
            "description": null,
            "generatedName": "ComponentsSchemasUnionJobJobTlInference",
            "groupName": null,
            "nameOverride": null,
            "schema": "UnionJob_Job<TlInference>",
            "type": "reference",
          },
          {
            "availability": null,
            "description": null,
            "generatedName": "ComponentsSchemasUnionJobJobTraining",
            "groupName": null,
            "nameOverride": null,
            "schema": "UnionJob_Job<Training>",
            "type": "reference",
          },
        ],
        "type": "undisciminated",
      },
    },
  },
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Hume AI Batch API",
  "variables": {},
  "webhooks": [],
}
`;
